### ISSUE TITLE
search and timeline places shown despite being hidden in Dolphin

### ISSUE THREAD

**Issue by ysalmon** [dfaust/plasma-applet-places-widget#3]
I have hidden all of the timeline and search predefined places in Dolphin, but they still show up in your widget. Furthermore I would like to only display disks that are not mounted by /etc/fstab (as those are integrated within the main filesystem).

I forked your repo to tweak the widget to my needs and maybe propose a PR but I am stuck : while QML seems pretty straighforward to understand and while I was able to find the documentation for PlasmaCore and its DataSource class, I was not able to find the documentation for the "places" source that you use and therefore I have no idea of the name of the properties I have to check to achieve what I want to do.

Do you have any pointers ? I find it extremely difficult to search the web for this information as "places" and "plasma" are very common words.

**Comment by dfaust** [comment]
> I was not able to find the documentation for the "places" source

You can use the `plasmaengineexplorer` to get the "model" of the "places" source (right-click, get model). But the data it provides is limited.

**Comment by ysalmon** [comment]
Thanks. Indeed, information is scarce and not detailed. But one can guess.
I made an [attempt](https://github.com/dfaust/plasma-applet-places-widget/pull/4). In my feeble understanding of QML, this should work. It does not. Possible causes are described there.

BTW, is there an easier way to preview/debug a widget than removing it and re-adding it to my panel ?

**Comment by dfaust** [comment]
> BTW, is there an easier way to preview/debug a widget than removing it and re-adding it to my panel ?

Yes, you can use `plasmoidviewer --applet org.kde.placesWidget`.

The widget must be installed first:
```
git clone https://github.com/dfaust/plasma-applet-places-widget
cd plasma-applet-places-widget
mkdir build
cd build
cmake -DCMAKE_INSTALL_PREFIX=/usr ..
make
sudo make install
```

**Comment by amithiel** [comment]
Hey there, is there any progress on this matter? the "search" entries are really annoying , it doubles the menu size

**Comment by dfaust** [comment]
I just released version 1.3 of `plasma-applet-places-widget` with the ability to filter timeline and search items. Please let me know if you find any bugs.
